src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Lists/BitBasicListTests.cs

```diff
@@ -0,0 +1,134 @@
+
+            int comItemCount = bitList.ChildElementCount;
+            if (virtualize)
+            {
+                Assert.AreEqual(comItemCount, itemCount + 2);
+            }
+            else
+            {
+                Assert.AreEqual(comItemCount, itemCount);
+            }
+        }
+
+        [DataTestMethod,
+            DataRow(Visual.Fluent, 1000, true, 1),
+            DataRow(Visual.Fluent, 100, false, 1),
+            
+            DataRow(Visual.Cupertino, 1000, true, 1),
+            DataRow(Visual.Cupertino, 100, false, 1),
+            
+            DataRow(Visual.Material, 1000, true, 1),
+            DataRow(Visual.Material, 100, false, 1),
+            ]
+        public void BitBasicListRenderExtraItemsUsingOverScanCount(Visual visual,
+             int itemCount, bool virtualize, int overScanCount)
+        {
+            var component = RenderComponent<BitBasicListTest>(parameters =>
+            {
+                parameters.Add(p => p.Visual, visual);
+                parameters.Add(p => p.Virtualize, virtualize);
+                parameters.Add(p => p.Items, GetTestData(itemCount));
+                parameters.Add(p => p.OverscanCount, overScanCount);
+            });
+
+            var bitList = component.Find(".bit-bsc-lst");
+
+            var visualClass = visual == Visual.Cupertino ? "cupertino" : visual == Visual.Material ? "material" : "fluent";
+            Assert.IsTrue(bitList.ClassList.Contains($"bit-bsc-lst-{visualClass}"));
+
+            int comItemCount = bitList.ChildElementCount;
+            if (virtualize)
+            {
+                Assert.AreEqual(comItemCount, itemCount + (overScanCount + 2));
+            }
+            else
+            {
+                Assert.AreEqual(comItemCount, itemCount);
+            }
+        }
+
+        [DataTestMethod,
+            DataRow(Visual.Fluent, 1000, "RoleItem", true),
+            DataRow(Visual.Fluent, 1000, "RoleItem", true),
+
+            DataRow(Visual.Cupertino, 1000, "RoleItem", true),
+            DataRow(Visual.Cupertino, 1000, "RoleItem", true),
+
+            DataRow(Visual.Material, 1000, "RoleItem", true),
+            DataRow(Visual.Material, 1000, "RoleItem", true),
+            ]
+        public void BitBasicListCheckForCorrectRole(Visual visual, int itemCount, string role, bool virtualize)
+        {
+            var component = RenderComponent<BitBasicListTest>(parameters =>
+            {
+                parameters.Add(p => p.Visual, visual);
+                parameters.Add(p => p.Items, GetTestData(itemCount));
+                parameters.Add(p => p.Role, role);
+                parameters.Add(p => p.Virtualize, virtualize);
+            });
+            var bitList = component.Find(".bit-bsc-lst");
+
+            var visualClass = visual == Visual.Cupertino ? "cupertino" : visual == Visual.Material ? "material" : "fluent";
+            Assert.IsTrue(bitList.ClassList.Contains($"bit-bsc-lst-{visualClass}"));
```

Comment:
this assertion is not related to this test. please move it to its own test describing the behavior of classes of the component